Objects/moduleobject.c

```diff
@@ -27,6 +27,26 @@ static PyMemberDef module_members[] = {
     {0}
 };
 
+static void
+assert_def_missing_or_redundant(PyModuleObject *m) {
+    /* We copy all relevant info into the module object.
+     * Modules created using a def keep a reference to that (statically
+     * allocated) def; the info there should match what we have in the module.
+     */
+#ifdef Py_DEBUG
```

Comment:
You can build Python in release mode with assertions:

```suggestion
#ifndef NDEBUG
```